src/briefcase/integrations/base.py

```diff
@@ -57,7 +57,7 @@ class Tool(ABC):
 
     name: str
     full_name: str
-    supported_host_os: set[str] = {"Darwin", "Linux", "Windows"}
+    supported_host_os: Collection[str] = {"Darwin", "Linux", "Windows"}
```

Comment:
hmm...the usefulness of [RUF012](https://docs.astral.sh/ruff/rules/mutable-class-default/) seems limited in BeeWare's source. Type correctness isn't enforced; so, this only helps if the developer's IDE escalates conflicts.

Specifically here, though, this isn't really an improvement; a `Collection` is just as mutable as a `set`.....maybe if we made these all `frozensets`.

The usefulness is most present for `dict` to `Mapping`, though; at least, the `Mapping` type implies immutability.

Reply:
idk...i think it is clever insofar as achieving the stated purpose of the ruff rule. It isn't inherently problematic and in lieu of a better non-typing solution, I guess it's ultimately fine.

Reply:
Yeah - I'm definitely keeping an eye on ty to see if it becomes a viable option.